lib/repl.js

```diff
@@ -386,14 +386,15 @@ function REPLServer(prompt,
   self.bufferedCommand = '';
   self.lines.level = [];
 
-  function complete(text, callback) {
-    self.complete(text, callback);
-  }
+  // Figure out which "complete" function to use.
+  self.completer = (typeof options.completer === 'function')
+    ? options.completer
+    : complete.bind(self);
```

Comment:
`complete()` was not bound, but the inner `self.complete` was, which I think it makes `complete.bind` a must to not broken its implementation, since it relies on `this`.

What can we further improve FICS and IMHO, is whether:

* custom `options.completer` needs to be binded too or not, since users could have access to useful REPL properties like `this.lines`, used in the default completer provided by the REPL.

* the `apply` is deemed as not needed if the params are directly coded in `REPL.prototype.complete`, which will require then some logic to pass to the `self.completer` the correct parameters since it accepts one for sync operation and two for async operation.